packages/styling/src/styles/theme.ts

```diff
@@ -13,11 +14,16 @@ import {
 } from './DefaultPalette';
 import { loadTheme as legacyLoadTheme } from '@microsoft/load-themed-styles';
 
+const defaultThemeFlags: IThemeFlags = {
+  disableGlobalClassNames: false,
```

Comment:
This should NOT be part of the theme, it's part of the global settings of the page. (In other words, it wouldn't make sense for this setting to change when you change the theme.)

Reply:
@phkuo Assuming we mix together themes, it makes sense to me. I'm a little confused. In a portion of the ui, we expect the styling to not have global class names. It feels like the right approach to me.

Reply:
What do you mean by "we mix together themes"?

The theme is independent of the CSS on the page. It doesn't know and it doesn't care about CSS. It should be portable between web and non-web platforms.

Let's say you have custom CSS on the page targeting global classes for a web part. It doesn't make sense that as you go from theme to theme, that web part randomly does or doesn't pick up your custom CSS. If you put custom global CSS on the page, you want to separately manage whether it applies or not independent of the theme. This setting should not be tied to the theme.

It should be treated like "theme" with @customizable, where you can mark portions of the page with it. Then you can turn it on/off globally, or just set it for one dialog or panel.

Reply:
@dzearing 
1.-3. seem like exactly the thing Customizer is for, no? Yes it does add bloat. I assumed any kind of global setting we'd want to be able to control it on a scoped basis as well. But I'm okay passing on it.

"Burden of knowledge" as in now every control has to have a const GlobalClassNames as well, on top of all the other stuff they already must do for getStyles(). I already have a two page document for myself on converting existing SCSS controls over to getStyles(), and I'm STILL adding to it. (I need to find time to polish and publish it.) I believe frameworks should remove boiler plate code, not add to it.
My proposed alternative is to put the logic in mergeStyles() (somewhere deep down in _processArgs()). If the passed in thing is a string, and it's not in our hash table, then throw it away.

I still STRONGLY believe it shouldn't be on the theme object. Example: you have a restaurant with a website. Part of the page is some list/table component that's the kids' menu. They're using global CSS to make the kids' menu Comic Sans, because it's a kids menu. Then they download some themes from online. Half the themes randomly make the kids' menu not respect the Comic Sans style, half of them do, with no rhyme or reason.
The theme should not contain settings.

Reply:
I think we should not have global classes at all. They are a relic of the past and create a mess of pollution, bugs, and frustration.

But this is a compromise. We don't want to break backwards compatibility for the things *we* define.

If a caller passes in a global classname, that's fine. So no, this is not something we would "disable in mergeStyles".

So the "burden of knowledge" does not impact the consumer. It is 100% transparent to them. I think you're only concern is inside of Fabric components? But again, think of this as a flight that is by default ON. We want to suppress them, but

Reply:
@phkuo I kind of see your point and i had the same thought as you at first, that we could maybe do this in mergeStyles, but it also seems like the wrong place to do this and not really related to it's responsibilities.

As David says, we are doing this as a temporary solution and we would eventually like to remove global styles altogether, when that happens we can remove this flag from the theme again.

Additionally making people more aware when they are adding global styles to by having them write the boiler plate (in the `component.styles.ts files`) might actually be a good thing as then they might reconsider if they actually need them in the first place ;)